### ISSUE TITLE
golangci-lint linter: automatically infer the execution directory

### ISSUE THREAD

**Issue by CarlJi** [qiniu/reviewbot#207]
background:

```bash
😱🚀 Linter: golangci-lint 
Repo: goplus/builder 
PR:   https://github.com/goplus/builder/pull/615 
Event: d4605c30-3469-11ef-83a5-c58febdb5641 
Unexpected: level=error msg="[linters_context] typechecking error: pattern ./...: directory prefix . does not contain main module or its selected dependencies"
```
In some monorepos, Go code is not placed in the main directory, and the current golangci-lint cannot automatically discover it.
Reviewbot currently supports custom workDir, but it would be better if it could be automatically determined.

**Comment by CarlJi** [comment]
FYI: https://github.com/golangci/golangci-lint/issues/825

**Comment by CarlJi** [comment]
fixed by #235
